### ISSUE TITLE
Including matcher in 'handle_response' line has confusing error

### ISSUE THREAD

**Issue by foxt** [caddyserver/caddy#6529]
If you use the following Caddyfile:
```caddyfile
reverse_proxy localhost:8000 {
    handle_response header Foo {
        # do something
    }
}
```

you will get the following error: `adapting config using caddyfile: parsing caddyfile tokens for 'reverse_proxy': configuring 'handle_response' for status code replacement is no longer supported. Use 'replace_status' instead.` even though you're not touching the status code.

If you use either 
```caddyfile
handle_response header Foo * {
```
or
```caddyfile
@bad status 500
handle_response @bad {
```
no error is raised.

Seems to be a check implemented at https://github.com/caddyserver/caddy/blob/54a0c8f94821f393edec1a1bc9459f2363c95117/modules/caddyhttp/reverseproxy/caddyfile.go#L915 that just complains if there are two parameters passed to `handle_response`.

**Comment by mholt** [comment]
Ah thanks.

@francislavoie This was from about 2 years ago; maybe it's time to revisit while we're at it?

**Comment by TowyTowy** [comment]
Happy to pick this up with exactly the change described above — reject more than a single matcher argument and drop the misleading `replace_status` reference from the error message. I see #7709 is still open but it's been idle a while and took a different direction than what was suggested; if the author is still on it I'll stay out of the way, otherwise I can have a small PR (with an adapt test) up shortly.

**Comment by steadytao** [comment]
> it's been idle a while and took a different direction than what was suggested; if the author is still on it I'll stay out of the way, otherwise I can have a small PR (with an adapt test) up shortly.

Yes, please go ahead; just make sure to keep the new PR focused on rejecting more than one matcher argument, removing the obsolete and misleading `replace_status` explanation and adding an adaptation test for the reported case please.
